spacify-engine/src/main/java/org/uci/spacifyEngine/services/AvailableSlotsService.java

```diff
@@ -0,0 +1,80 @@
+package org.uci.spacifyEngine.services;
+
+import com.fasterxml.jackson.databind.util.JSONPObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.uci.spacifyLib.entity.AvailableSlotsEntity;
+import org.uci.spacifyLib.entity.ReservationEntity;
+import org.uci.spacifyLib.entity.RoomEntity;
+import org.uci.spacifyLib.entity.RoomType;
+import org.uci.spacifyLib.repsitory.AvailableSlotsRepository;
+import org.uci.spacifyLib.repsitory.ReservationRepository;
+import org.uci.spacifyLib.repsitory.RoomRepository;
+
+import java.math.BigInteger;
+import java.sql.Time;
+import java.time.LocalDateTime;
+import java.util.*;
+
+@Service
+public class AvailableSlotsService {
+
+    @Autowired
+    private RoomRepository roomRepository;
+
+    @Autowired
+    private ReservationRepository reservationRepository;
+
+    @Autowired
+    private AvailableSlotsRepository availableSlotsRepository;
+
+    public List<ReservationEntity> getAllAvailableSlots() {
+        String room_type = "STUDY";                                                     //to take from UI
```

Comment:
Don't hard code this. You need to take room_type as parameter in this method, which will be passed to the API in the controller from UI. Check the User profile api in the UserController for reference.

Reply:
Same for the fromTime and toTime values: pass them as parameters